flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/DefaultExecutionSlotAllocatorTest.java

```diff
@@ -200,14 +203,16 @@ private SlotExecutionVertexAssignment findSlotAssignmentByExecutionVertexId(
 						executionVertexId)));
 	}
 
-	private static class AllocationToggableSlotProvider implements SlotProvider {
+	static class AllocationToggableSlotProvider implements SlotProvider {
 
 		private final List<Tuple3<SlotRequestId, ScheduledUnit, SlotProfile>> slotAllocationRequests = new ArrayList<>();
 
 		private final List<SlotRequestId> cancelledSlotRequestIds = new ArrayList<>();
 
 		private boolean slotAllocationDisabled;
 
+		private boolean forceFailingSlotAllocation = false;
```

Comment:
```suggestion
		private boolean forceFailingSlotAllocation;
```

Reply:
nit: it would be nice to introduce `enum AllocationTypeEnum {ENABLED, DISABLED, FAILING}`

Reply:
`OneSlotPerExecutionSlotAllocatorTest` now has its own `TestingBulkSlotProvider` which does not have 3 modes.